internal/middleware/preload_schema.go

```diff
@@ -0,0 +1,123 @@
+package middleware
+
+import (
+	"fmt"
+	"github.com/go-kratos/kratos/v2/log"
+	"gopkg.in/yaml.v3"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+var schemaCache sync.Map
+
+func PreloadAllSchemas(resourceDir string) error {
+	// Set default resource directory if not provided
+	if resourceDir == "" {
+		resourceDir = "data/schema/resources"
```

Comment:
I'm not a fan of hardcoding paths so deep inside the codebase. Can we ensure the default is defined at the config level?

Reply:
I agree, we can make it configurable.